pset-iterators.js

```diff
@@ -184,6 +246,38 @@ console.log('Problem 7:')
 
 // Add your code below this line
 
+let tookOfficeYear = 0
+let leftOfficeYear = 0
+
+const shortTermPresidents = presidents.filter(function (president){
+
+  tookOfficeYear = president.took_office.substring(0, 4)
+// console.log("--------------")
+// console.log(president.took_office.substring(0, 4))
+
+  if (president.left_office === null) {
+      var d = new Date();
+      president.left_office = d.getFullYear();
+//      console.log(president.left_office)
+
+  } else {
+      leftOfficeYear = president.left_office.substring(0, 4)
+//    console.log(president.left_office)
+  }
+
+  if (leftOfficeYear - tookOfficeYear < 4) {
+
+    return president
+
+  } else {
+//   console.log("less than for years")
+   return
+  }
+})
+
+console.log(shortTermPresidents)
+
+
```

Comment:
@thetripler Good Job on this assignment here using the filter method to return the president who has served for less than a 4-year term. One suggestion since we have one president whose left office value is null we should leave the president off the list for this solution.

Take a look at another approach [here](https://github.com/jsr20190826/hw-03-iterators-solution/blob/96bab20deba3ad116374a5827291a951faf5d555/pset-iterators.js#L201)